samples/smpe_list.py

```diff
@@ -0,0 +1,181 @@
+//******************************************************************
+//SMPLST   EXEC PGM=GIMSMP
+//SMPCSI   DD  DSN=AS4SMP.GLOBAL.CSI,DISP=SHR
+//SMPLOG   DD  DUMMY
+//SMPLOGA  DD  DUMMY
+//SMPWRK6  DD DSN=&&TEMP,
+//       SPACE=(CYL,(2500,100,500)),VOL=SER=USRAT5,UNIT=3390,
+//       DCB=(DSORG=PO,RECFM=FB,LRECL=80,BLKSIZE=3200),
+//       DISP=(NEW,DELETE)
+//SMPCNTL  DD  *
+ SET BDY(GLOBAL).
+ LIST.
+ '''
+import sys
+import os
+import yaml
+import argparse
+from zoautil_py import mvscmd, datasets
+from zoautil_py.types import DDStatement, DatasetDefinition, FileDefinition
+
+
+def SMPE_list(target_zone="GLOBAL", list_options=None,
+              high_level_qualifier="SYS1"):
+    '''
+    This function does the heavy lifting. It performs the function that
+        is contained in the JCL. It sets up the DD statements and issues the
+        call to the executable.
+    '''
+
+    # Initialize DD List
+    dd_list = []
+
+    # and the temporary dataset
+    temp_dataset = None
+
+    # get the defaults from the yaml file
+    defaults = getdefaults("./SMPElistDefaults.yaml")
+
+    try:
+        # Setup base DDs
+        dd_list.append(DDStatement("SMPCSI",
+                                   DatasetDefinition(
+                                       defaults["SMPECSI"]["Dataset"])))
+        dd_list.append(DDStatement("SMPLOG", "DUMMY"))
+        dd_list.append(DDStatement("SMPLOGA", "DUMMY"))
+
+        # Create Temporary File
+        temp_dataset_name = datasets.tmp_name(high_level_qualifier)
+        temp_dataset = datasets.create(temp_dataset_name, type="PDS",
+                                       primary_space=(defaults["TEMP_DATASET"]["Primary_space"]).strip(),
+                                       secondary_space=(defaults["TEMP_DATASET"]["Secondary_space"]).strip(),
+                                       block_size=3200,
+                                       record_format="FB", record_length=80,
+                                       volumes=(defaults["TEMP_DATASET"]["Volume"]).strip(),
+                                       directory_blocks=10)
+
+        # add it to the ddList
+        dd_list.append(DDStatement("SMPWRK6",
+                                   DatasetDefinition(temp_dataset_name)))
+
+        # define the input for the program make - sure the input is EBCIDIC
+        sysin_file_name = defaults["SMPECNTL"]["Filename"]
+        with open(sysin_file_name, mode="w", encoding="cp1047") as file:
+            file.write(f"SET     BDY({target_zone}).\n")
+            if list_options is None:
+                file.write("LIST.\n")
+            else:
+                file.write(f"LIST {list_options}.\n")
+        dd_list.append(DDStatement("SMPCNTL",
+                                   FileDefinition(sysin_file_name)))
+
+        # define the place for the output to go
+        output_dataset_name = datasets.tmp_name(high_level_qualifier)
+        _ = datasets.create(output_dataset_name, type="SEQ",
```

Comment:
Why is this being written to a throwaway variable?

Reply:
The output of the create is unused. If there is an error it is handled by the except block. The dataset pointer is the name of the dataset.

Reply:
I get that, but I don't get why you're assigning it to something. Why not just make the dot function call on the `datasets` variable without storing it to the throwaway variable. For instance.

If it's unused why this:
```
_ = datasets.create(output_dataset_name, type="SEQ",
                    primary_space=(defaults["OUTPUT_DATASET"]["Primary_space"]).strip(),
                    secondary_space=(defaults["OUTPUT_DATASET"]["Secondary_space"]).strip(),
                    volumes=(defaults["OUTPUT_DATASET"]["Volume"]).strip())
```

Instead of this:
```
datasets.create(output_dataset_name, type="SEQ",
                primary_space=(defaults["OUTPUT_DATASET"]["Primary_space"]).strip(),
                secondary_space=(defaults["OUTPUT_DATASET"]["Secondary_space"]).strip(),
                volumes=(defaults["OUTPUT_DATASET"]["Volume"]).strip())
```